### DISCUSSION TITLE
Functional test doesn't seem to be working in SFRA storefront-reference-architecture

### DISCUSSION THREAD

**Discussion by steffy** [SalesforceCommerceCloud/sfcc-ci#401] [Team Posts]
<img width="475" alt="SFRA" src="https://user-images.githubusercontent.com/2849679/58603793-2e286900-82e6-11e9-806d-d534bfa74971.PNG">

Trying to run the existing tests. 

Thanks

**Comment by zlatinz** [comment]
Do you get the same error for vanilla SFRA code?

**Comment by echessman** [comment]
@steffy While I have not run into that particular error I do know that the SFRA team's functional test are broken. 

Writing functional UI test were not part of the SFRA team's definition of done for stories and bugs. As a result the functional UI test were left unattended and have been broken for some time. The SFRA team recognizes the importance of these tests and has since altered our definition of done for stories to include functional UI test. In the next version of SFRA, we have removed the existing functional UI test and will be introducing codeceptJS  as the new framework for our functional UI test. At this time, we currently do not have many tests written with this new framework, however we do have a priority on getting more functional UI tests written to help set a baseline for SFRA. We are sorry for the inconvenience and thank you for your patience.

**Comment by steffy** [comment]
> Do you get the same error for vanilla SFRA code?

No. this is only for the functional tests.

**Comment by steffy** [comment]
> @steffy While I have not run into that particular error I do know that the SFRA team's functional test are broken.
> 
> Writing functional UI test were not part of the SFRA team's definition of done for stories and bugs. As a result the functional UI test were left unattended and have been broken for some time. The SFRA team recognizes the importance of these tests and has since altered our definition of done for stories to include functional UI test. In the next version of SFRA, we have removed the existing functional UI test and will be introducing codeceptJS as the new framework for our functional UI test. At this time, we currently do not have many tests written with this new framework, however we do have a priority on getting more functional UI tests written to help set a baseline for SFRA. We are sorry for the inconvenience and thank you for your patience.

@echessman Thanks for the update. Are you able to give me a little more detail on the new framework? We are about to start our project next week. We were thinking of using the same framework as SFRA tests for our UI tests.
Cheers,

**Comment by echessman** [comment]
@steffy Sorry for the delayed response. I personally have not worked with CodeceptJS. To get more information you may want to look at the Codecept documentation and the SFRA code base.

**Comment by nicekiwi** [comment]
Can you share any details around the next release? ETA, Documentation?

**Comment by echessman** [comment]
The SFRA team released version 4.2.0 and can be found here https://github.com/SalesforceCommerceCloud/storefront-reference-architecture/releases/tag/v4.2.0
